### PR TITLE
feat: add span for showing operation_id in state machine transition

### LINKED ISSUE #3883: State transitions/trigger futures should be executed inside spans
The span should contain the operation id, so that logs from state transitions and trigger futures can be followed based on that.

### PR DESCRIPTION
**Pull request by Maan2003**
closes #3883

### CODE REVIEW

File: fedimint-client/src/sm/executor.rs [hunk 1]
```diff
@@ -434,102 +434,113 @@ where
...
-            debug!(
-                ?state,
-                transition_outcome = ?AbbreviateJson(&transition_outcome),
-                "Executing state transition"
-            );
+            let operation_id = state.operation_id();
+            // info level because spans only log when any event happens within a span.
+            let span = tracing::info_span!("state_machine_transition", %operation_id);
+            async {
+                // TODO: add a INFO version without all the details
```
Comment (elsirion) [inline fedimint-client/src/sm/executor.rs:441]: Still relevant?

**Review by elsirion — APPROVED** [review]
Just one minor nit

File: fedimint-client/src/sm/executor.rs [hunk 2]
```diff
@@ -434,102 +434,113 @@ where
...
+                info!(
```
Comment (Maan2003) [inline fedimint-client/src/sm/executor.rs:442]: oh I made this acccidentally info

**Review by elsirion — APPROVED** [review]

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.2`:
- #3922

**Pull request merged**